include/exadg/incompressible_navier_stokes/postprocessor/output_generator.cpp

```diff
@@ -200,6 +257,13 @@ OutputGenerator<dim, Number>::evaluate(
                     additional_fields,
                     time_control.get_counter(),
                     mpi_comm);
+
+  if(dim == 3 && surface_fields.size() > 0)
```

Comment:
I currently think this if statement should appear inside the function `write_surface_output()` from a design perspective.